scripts/build_utils.sh

```diff
@@ -837,6 +837,16 @@ gen_debian_version() {
     echo "${raw}${bptag}"
 }
 
+setup_cmakeoptions_from_ci() {
+  if [-z BRANCH ];then
+    curl "https://raw.githubusercontent.com/ceph/ceph-ci/$BRANCH/CMakeLists.txt" --output CMakeLists.txt.$BRANCH
```

Comment:
A couple things here.

1. It's not necessary.  The Jenkins job has already cloned the ceph repo and checked out whatever branch is being built.  Just `grep -q "^WITH_JAEGER.*ON" CMakeLists.txt`.
2. You're running this function in `ceph-dev-setup` which builds the tarballs, etc. for branches in ceph.git.  But you're attempting to `curl` a file from ceph-ci.git.

ceph-ci.git:BRANCH != ceph.git:BRANCH.

Additionally, ceph-ci.git doesn't have pacific, octopus, etc. which is all the ceph-dev pipeline builds.

ceph-dev = ceph.git
ceph-dev-new = ceph-ci.git

Reply:
i'd suggest just conditionalize on `BRANCH`, instead of checking a certain file in repo.

Reply:
thank you for the suggestions! updated: https://github.com/ceph/ceph-build/pull/1862/files#diff-1326d7a2d74cace6f00e1cd5d3ad15ead3fc6359eed098d2133fd72c0ec6700bR840